### PR TITLE
Proposed new intro incl. a bit of re-org

### PR DESCRIPTION
**Pull request by mirjak**
I'm proposing this change because I think this new intro summarises better the main intention of this document. Discussing the robustness principle right at the start of the intro is rather distracting and now feels more like a bit of background knowledge for me given how the scope of the document changes over time, so I propose to also have all that discussion as subsections of the intro rather than own main session.

Plus some minor modifications where the phrase "application of the robustness principle" is used rather broadly while more specifically "silently dropping unexpected input" is meant. This should really help to clarify. There might still be more such cases.

### CODE REVIEW

File: draft-iab-protocol-maintenance.md [hunk 1]
```diff
@@ -60,6 +60,27 @@ maintained, protocol designers and implementers can avoid these pitfalls.
...
+The main goal of the networking standards process is to enable the long term
+interoperability of protocols. This document describes active protocol
+maintenance, a means to accomplish that goal. By evolving specifications and
+implementations, it is possible to reduce ambiguity over time and create a
+healthy ecosystem.
+
+There is good evidence to suggest that many important protocols are routinely
+maintained beyond their inception. In particular, a sizeable proportion of IETF
+activity is dedicated to the stewardship of existing protocols. This document
+fisrt discusses hazards in applying the robustness principle too broadly in
```
Comment (mirjak) [inline draft-iab-protocol-maintenance.md:72]:
```suggestion
fisrt discusses hazards in applying the robustness principle too broadly in
```

File: draft-iab-protocol-maintenance.md [hunk 2]
```diff
@@ -107,15 +128,19 @@ ambiguous message be processed according to the inferred intent of the sender.
 The facets of the robustness principle that protect against defects or attack
 are understood to be necessary guiding principles for the design and
 implementation of networked systems. However, an interpretation that advocates
-for tolerating unexpected inputs is no longer considered best practice in all
-scenarios.
+for tolerating unexpected inputs is no longer recommended in this document,
```
Comment (mirjak) [inline draft-iab-protocol-maintenance.md:131]:
```suggestion
for tolerating unexpected inputs is no longer recommended in this document,
```

File: draft-iab-protocol-maintenance.md [hunk 3]
```diff
@@ -107,15 +128,19 @@ ambiguous message be processed according to the inferred intent of the sender.
...
-for tolerating unexpected inputs is no longer considered best practice in all
-scenarios.
+for tolerating unexpected inputs is no longer recommended in this document, 
+instead whereever possible active maintenance to enable quic updates should
+be applied.
 
 Time and experience shows that negative consequences to interoperability
-accumulate over time if implementations silently accept faulty input. This
-problem originates from an implicit assumption that it is not possible to effect
+accumulate over time if implementations silently accept faulty input.
```
Comment (mirjak) [inline draft-iab-protocol-maintenance.md:136]:
```suggestion
accumulate over time if implementations silently accept faulty input.
```

File: draft-iab-protocol-maintenance.md [hunk 4]
```diff
@@ -221,42 +244,28 @@ former, interoperation depends on specifications that capture all relevant
...
+protocol maintenance and evolution. Employing this guidance is therefore only
+applicable where there is the possibility of improving deployments through
+timely updates of their implementations.
 
-This premise that specifications will be imperfect is correct. However, ignoring
-faulty or ambiguous input is almost always the incorrect solution to the problem.
 
 
-# Harmful Consequences of Tolerating the Unexpected
+# Harmful Consequences of Tolerating the Unexpected
```
Comment (mirjak) [inline draft-iab-protocol-maintenance.md:268]:
```suggestion
# Harmful Consequences of Tolerating the Unexpected
```

**Review by martinthomson — COMMENTED** [review]
I find the restructuring a little hard to follow.  I don't mind the part where text is moved up into the introduction, but there are a few additions as things go that I have a harder time with.

Also, because this is not just a rearrangement, it's quite hard to review.  That said, I think that I like this ordering, if we lift most of the heading levels here up one. 

@DavidSchinazi, does this work for you?

1. Intro
2. Protocol Robustness
  2.1 Fallibility of Specifications
  2.2 Extensibility
  2.3 Flexible Protocols
4. Applicability
5. etc...

**Comment by mirjak** [comment]
Please go ahead and apply any changes you like. This was mainly to better show what my comments were about. I know with the restructuring is always a bit hard to read the diff. I tried to keep it at a minimum.

**Comment by martinthomson** [comment]
Merged in pieces with #77 and #78.
